echopop/nwfsc_feat/apportion.py

```diff
@@ -1,123 +1,704 @@
+    3. Creates pivot tables from proportions data
+    4. Multiplies aggregated mesh values by proportions
+    5. Returns results in the same format as input proportions
+
+    The distribution preserves the biological group structure while scaling values
+    according to the proportion data provided.
+    """
+
+    # Convert to compatible format
+    if isinstance(proportions, pd.DataFrame):
+        proportions = {"distributed_mesh": proportions}
+
+    # Verify that `variable` exists
+    if variable not in mesh_data_df.columns:
+        raise ValueError(f"Variable '{variable}' not found in `mesh_data_df`.")
+
+    # Sum variables over indices
+    mesh_data_pvt = mesh_data_df.rename(columns=mesh_proportions_link).pivot_table(
+        index=stratify_by, values=variable, aggfunc="sum", observed=False
+    )
+
+    # Parse the additional columns that are required for grouping
+    proportions_group_columns = {
+        k: [c for c in (list(v.index.names) + list(v.columns)) if c in group_by]
+        for k, v in proportions.items()
+    }
+
+    # Convert to DataFrame(s) to pivot table(s)
+    proportions_grouped_pvt = {
+        k: (
+            df
+            if utils.is_pivot_table(df)
+            else utils.create_pivot_table(
+                df,
+                index_cols=proportions_group_columns[k],
+                strat_cols=stratify_by,
+                value_col="proportion_overall",
+            )
+        )
+        for k, df in proportions.items()
+    }
+
+    # Distribute the variable over each table
+    apportioned_grouped_pvt = {
+        k: df.mul(mesh_data_pvt[variable]).fillna(0.0) for k, df in proportions_grouped_pvt.items()
+    }
+
+    # Return output in the same format as the proportions input
+    return apportioned_grouped_pvt
+
+
+def impute_kriged_table(
+    reference_table_df: pd.DataFrame,
+    initial_table_df: pd.DataFrame,
+    standardized_table_df: pd.DataFrame,
+    group_by: List[str],
+    impute_variable: List[str],
+    table_reference_indices: List[str],
+) -> pd.DataFrame:
+    """
+    Perform nearest-neighbor imputation of missing values in population apportionment tables.
+
+    This function fills missing values in standardized population tables by finding the nearest
+    non-zero reference values and using them to impute proportional values based on the
+    reference table structure.
+
+    Parameters
+    ----------
+    reference_table_df : pd.DataFrame
+        Reference table used to determine which values should be imputed and to calculate
+        imputation ratios. Typically contains aged fish data with complete age structure.
+    initial_table_df : pd.DataFrame
+        Original population table before standardization. Used to determine which cells
+        have actual data vs. missing values.
+    standardized_table_df : pd.DataFrame
+        Standardized population table that needs imputation. This will be modified with
+        imputed values.
```

Comment:
What do you mean by "standardized" here? Do you mean the total sum to 1 along one or more of the dimensions?